### ISSUE TITLE
How to document functions passed as arguments ?

### ISSUE THREAD

**Issue by estebanzimanyi** [doxygen/doxygen#10136]
For example, using the example of a priority queue
```
/**
 * @brief Create a new priority queue
 * @param cmp Compare function that returns
 * - 0 if d1 and d2 have the same priorities
 * - [negative value] if d1 have a smaller priority than d2
 * - [positive value] if d1 have a greater priority than d2
*/
PQueue *pqueue_make(int (*cmp)(const void *d1, const void *d2))
{
  ...
}
```
I receive the warning
```
warning: argument 'cmp' from the argument list of pqueue_make has multiple @param documentation sections
```
How to properly document such functions passed as arguments ?

**Comment by albert-github** [comment]
With the simple example as given, with a default Doxyfile  and using the current doxygen version (1.9.7) I don't get a warning

So:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by estebanzimanyi** [comment]
You are right !
I am having problems to reproduce the error using a self-contained example, the error should be somewhere else ...
While looking for the solution, actually my question would be what is the best way to document both 
* the arguments and the result of the "external" function
* the arguments and the result of the argument function
as in the following example. 
```
/**
 * @brief Create a new priority queue
 * @param size Initial size of the queue
 * @param cmp Compare function that returns
 * - 0 if d1 and d2 have the same priorities
 * - [negative value] if d1 have a smaller priority than d2
 * - [positive value] if d1 have a greater priority than d2
*/
int *pqueue_make(size_t size,
  int (*cmp)(const void *d1, const void *d2))
{
  /* empty function */
  return 0;
}
*/
``` 
Should this be done with a @typedef, @fn or similar ? 
Thanks for your advice.

**Comment by albert-github** [comment]
Not an easy problem. We have here something nested.
I think best is to create some `typedef` or something similar with the `int (*cmp)(const void *d1, const void *d2)`  definition and use the `typedef` as type in the function definition of `pqueue_make`.

**Comment by Albrecht-S** [comment]
> I think best is to create some typedef or something similar with the int (*cmp)(const void *d1, const void *d2) definition ...

I agree that using a typedef for the function is a good solution. It is also less confusing for the reader. We are using it in the FLTK project (fltk.org, see issue #10295). However, the syntax similar to the one given in the example above causes issues in doxygen which is why I posted issue #10295.
